apply/apply.go

```diff
@@ -416,6 +416,7 @@ func applyEnvs(
 						Outputs:      componentPlan.Outputs,
 						Prefix:       nil,
 						ProvidersMap: componentPlan.ProvidersMap,
+						DependsOn:    componentPlan.DependsOn,
 					},
```

Comment:
This `if` is only used for the legacy case where the componentPlan is using a single `moduleSource`.. but you didn't add it to the `config` struct for this..

Reply:
for the case where the component takes a `module: [ .. , .. ]` array, the `ModuleInvocation` struct is directly passed down

https://github.com/vincenthsh/fogg/blob/08d7e9a6f2ccb4c62b6266bef8d5e4430d314f2e/apply/apply.go#L424-L437

Reply:
my bad, can't depend on any other module in a single module component 🤦‍♂️

Reply:
Drop this, it's different purpose